### PR TITLE
Release 3.6.0 and removal of TfsDataFrame methods

### PR DESCRIPTION
**Pull request by fsoubelet**
This is a follow-up of PR #119 

This is a new version, `3.6.0`, which removes the `append` and `join` compatibility methods from the `TfsDataFrame` object. Users should now use `tfs.frame.concat` instead.

This restores compatibility with `pandas` 2.x versions where the methods had been removed from `pandas.DataFrame`, and thus the version constraint on `pandas` for the package is returned to its previous value of `>=1.0` (as in `3.5.1`.)

The tests of said methods have been removed.

Naturally coverage decreases a bit since less lines of code are run.

### CODE REVIEW

**Comment by fsoubelet** [comment]
@JoschD in [25764d3](https://github.com/pylhc/tfs/pull/120/commits/25764d31a921428bcc50769be0a501facdc5f12a) I add `check_index_type=False` to an assert for empty index dataframe read/write.

This is because starting pandas 2.0 the index is properly read as empty, but the inferred type is `integer` instead of `empty` for the reference df. I think this is totally fine and we can adapt the test, but I'd like confirmation that you agree.

File: tests/test_frame.py [hunk 1]
```diff
@@ -60,20 +92,6 @@ def test_headers_merging_none_returns_empty_dict(self, _tfs_file_x_pathlib, _tfs
...
-        assert dframe_x.append(other=dframe_y, new_headers={}).headers == OrderedDict()
-        assert dframe_y.append(other=dframe_x, new_headers={}).headers == OrderedDict()
-
-        # we provide lsuffix (or rsuffix) since dframes have the same columns
-        assert dframe_x.join(other=dframe_y, lsuffix="_l", new_headers={}).headers == OrderedDict()
-        assert dframe_y.join(other=dframe_x, lsuffix="_l", new_headers={}).headers == OrderedDict()
-
-        assert dframe_x.merge(right=dframe_y, new_headers={}).headers == OrderedDict()
-        assert dframe_y.merge(right=dframe_x, new_headers={}).headers == OrderedDict()
-
```
Comment (JoschD) [inline tests/test_frame.py:76]: this test can stay, you only need to remove the lines until 67-73

Comment (fsoubelet) [reply tests/test_frame.py:76]: Indeed with `merge` back I will bring it back too

Comment (JoschD) [reply tests/test_frame.py:76]: and maybe add concat?

Comment (fsoubelet) [reply tests/test_frame.py]: Done!

**Pull request merged**